src/main/webapp/creatives/maps/google_maps_client.js

```diff
@@ -0,0 +1,269 @@
+import {CreativeMessage} from '../constants.js';
+const DEFAULT_MAP_LAT = 37.422004;
+const DEFAULT_MAP_LNG = -122.081402;
+const DEFAULT_ZOOM = 13;
+const DEFAULT_LOCATION_NUM_DISPLAYED = 4;
+const MARKER_SIZE = 25;
+
+/**
+ * This class handles all map related functionality, including
+ * displaying the map, calculating & displaying directions, tracking all
+ * interactions with the map, logging error messages with SimidProtocol,
+ * and handling any errors that result from the Maps API.
+ */
+export default class GoogleMapsClient {
+    /**
+     * A GoogleMapsClient object takes in a userSession object, a search query string, the marker image url,
+     * user coordinates, a SimidProtocol object, and a function that determine's the object's behavior in case
+     * of an error.
+     */
+    constructor(userSession, query, markerUrl,
+         coordinates = new google.maps.LatLng(DEFAULT_MAP_LAT, DEFAULT_MAP_LNG), simidProtocol, onMapsClientComplete) {
```

Comment:
make anything with a default value the last value. Otherwise it gets very confusing which variable you want to replace

Reply:
reorganized